src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritQueryHandlerHttp.java

```diff
@@ -0,0 +1,497 @@
+
+    if (getPatchSets) {
+      //ALL_REVISIONS: describe all revisions, not just current.
+      str.append("&o=ALL_REVISIONS");
+    }
+    if (getCurrentPatchSet) {
+      //.append(" --current-patch-set");
+      //CURRENT_REVISION: describe the current revision (patch set) of the change, including the commit SHA-1 and URLs
+      //to fetch from.
+      str.append("&o=CURRENT_REVISION");
+    }
+    if (getFiles && (getCurrentPatchSet || getPatchSets)) {
+      //str.append(" --files");
+      //CURRENT_FILES: list files modified by the commit and magic files, including basic line counts inserted/deleted
+      //per file.
+      // Only valid when the CURRENT_REVISION or ALL_REVISIONS option is selected.
+      str.append("&o=CURRENT_FILES");
+    }
+    if (getCommitMessage && (getCurrentPatchSet || getPatchSets)) {
+      //str.append(" --commit-message");
+      //CURRENT_COMMITS: parse and output all header fields from the commit object, including message.
+      // Only valid when the CURRENT_REVISION or ALL_REVISIONS option is selected.
+      str.append("&o=CURRENT_COMMIT");
+    }
+    if (getComments) {
+      //str.append(" --comments");
+      //MESSAGES: include messages associated with the change.
+      str.append("&o=MESSAGES");
+    }
+
+    logger.debug("sending: " + str);
+    //System.out.println("sending: " + str);
+
+    Response response = given().auth().preemptive()
+                        .basic(credential.getUserPrincipal().getName(), credential.getPassword())
+                        .given().when().request("GET", str.toString());
+
+    logger.debug("Status received: " + response.getStatusLine());
+
+    switch(response.statusCode()) {
+      case statusOk:
+        logger.debug("Body received: " + response.body().asPrettyString());
+        break;
+
+      case statusBadRequest:
+        logger.error(httpBaseUrl + ": Bad request " + "(400)");
+        throw new GerritQueryException(str + ": Bad request " + "(400)");
+
+      case statusBadCredentials:
+        logger.error("Unable to authenticate to \"" + httpBaseUrl + "\"");
+        throw new IOException("Error connecting to \"" + httpBaseUrl + "\": " + "(401)");
+
+      case statusNotFound:
+        logger.error("\"" + httpBaseUrl + "\": Could not be found! (404)");
+        throw new IOException("Error connecting to \"" + httpBaseUrl + "\": " + "(404)");
+
+      default:
+        logger.error("Error connecting to \"" + httpBaseUrl + "\"! (" + response.statusCode() + ")");
+        throw new IOException("Error connecting to \"" + httpBaseUrl + "\"! (" + response.statusCode() + ")");
+    }
+
+    //response.body().print();
+
+    //removing XSSI-Chars
+    String body = response.body().asString().split("\\r?\\n|\\r")[1];
+
+    JSONArray jsonArray = JSONArray.fromObject(body);
+
+    for (int i = 0; i < jsonArray.size(); i++) {
+      visitor.visit(jsonArray.getString(i));
```

Comment:
visitor.accept(jsonArray.getString(i));
die for Loop könnte man auch noch per Stream machen (kein muss).